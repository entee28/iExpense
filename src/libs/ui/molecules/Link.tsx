import React from 'react'
import { Pressable, PressableProps, Text } from '../atoms'
import colors from '../colors'

type LinkProps = {
  label: string
} & PressableProps

export const Link = (props: LinkProps) => {
  const { label, ...buttonProps } = props

  return (
    <Pressable {...buttonProps}>
      <Text
        testID="link_label"
        semiBold
        color={buttonProps.disabled ? colors.mono40 : colors.primary100}>
        {label}
      </Text>
    </Pressable>
  )
}
