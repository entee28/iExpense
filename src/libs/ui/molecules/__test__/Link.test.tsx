import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'
import { Link } from '../Link'

describe('Link', () => {
  it('correctly handles press event', () => {
    let pressed = false

    const press = () => {
      pressed = true
    }

    const link = render(<Link testID="link" label="Hello" onPress={press} />)

    fireEvent.press(link.getByTestId('link'))
    expect(pressed).toEqual(true)
  })

  it('should render label correctly', () => {
    const link = render(<Link label="Hello Test" testID="link" />)

    expect(link.getByTestId('link_label').children).toContain('Hello Test')
  })

  it('should match snapshot', () => {
    const snapshot = render(<Link label="Hello Test" testID="link" />).toJSON()

    expect(snapshot).toMatchSnapshot()
  })
})
