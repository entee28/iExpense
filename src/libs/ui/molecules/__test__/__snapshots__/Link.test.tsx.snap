// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="link"
>
  <Text
    style={
      [
        {
          "color": "#0055FF",
          "fontFamily": "Nunito-SemiBold",
          "fontSize": 14,
          "lineHeight": undefined,
          "textAlign": "auto",
        },
        {},
      ]
    }
    testID="link_label"
  >
    Hello Test
  </Text>
</View>
`;
